addons/account/migrations/13.0.1.1/pre-migration.py

```diff
@@ -0,0 +1,272 @@
+        for column, column_type in column_spec_list:
+            openupgrade.logged_query(
+                env.cr, """
+                ALTER TABLE {table}
+                ADD COLUMN {column} {column_type}""".format(
+                    table=table, column=column, column_type=column_type
+                ),
+            )
+
+
+def fill_account_move_line_parent_state(env):
+    """Faster way"""
+    openupgrade.logged_query(
+        env.cr, """
+        ALTER TABLE account_move_line
+        ADD COLUMN parent_state varchar""",
+    )
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE account_move_line aml
+        SET parent_state = am.state
+        FROM account_move am
+        WHERE am.id = aml.move_id""",
+    )
+
+
+def fill_account_move_line_account_internal_type(env):
+    """Faster way"""
+    openupgrade.logged_query(
+        env.cr, """
+        ALTER TABLE account_move_line
+        ADD COLUMN account_internal_type varchar""",
+    )
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE account_move_line aml
+        SET account_internal_type = aat.type
+        FROM account_account aa
+        JOIN account_account_type aat ON aa.user_type_id = aat.id
+        WHERE aml.account_id = aa.id""",
+    )
+
+
+def create_res_partner_ranks(env):
+    """Faster way"""
+    openupgrade.logged_query(
+        env.cr, """
+        ALTER TABLE res_partner
+        ADD COLUMN customer_rank integer
+        DEFAULT 0""",
+    )
+    openupgrade.logged_query(
+        env.cr, """
+        ALTER TABLE res_partner ALTER COLUMN customer_rank DROP DEFAULT""",
+    )
+    openupgrade.logged_query(
+        env.cr, """
+        ALTER TABLE res_partner
+        ADD COLUMN supplier_rank integer
+        DEFAULT 0""",
+    )
+    openupgrade.logged_query(
+        env.cr, """
+        ALTER TABLE res_partner ALTER COLUMN supplier_rank DROP DEFAULT""",
+    )
+
+
+def add_helper_invoice_move_rel(env):
+    openupgrade.logged_query(
+        env.cr, """
+        ALTER TABLE account_move
+        ADD COLUMN old_invoice_id integer""",
+    )
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE account_move am
+        SET old_invoice_id = ai.id
+        FROM account_invoice ai
+        WHERE ai.move_id = am.id
+        """,
+    )
+    index_name = '%s_%s_index' % ("account_move", "old_invoice_id")
+    sql.create_index(
+        env.cr, index_name, "account_move", ['"%s"' % "old_invoice_id"])
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE account_move am
+        SET old_invoice_id = aml.invoice_id
+        FROM account_move_line aml
+        WHERE aml.move_id = am.id AND am.old_invoice_id IS NULL
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE account_invoice ai
+        SET move_id = am.old_invoice_id
+        FROM account_move am
+        WHERE am.old_invoice_id = ai.id AND ai.move_id IS NULL
+        """,
+    )
```

Comment:
I think we can replace this just one query
<pre>
    openupgrade.logged_query(
        env.cr, """
        UPDATE account_invoice ai
        SET move_id = aml.move_id
        FROM account_move_line aml
        WHERE aml.invoice_id = ai.id AND ai.move_id IS NULL
        """,
    )
</pre>
but need write this before query:
<pre>
    openupgrade.logged_query(
        env.cr, """
        UPDATE account_move am
        SET old_invoice_id = ai.id
        FROM account_invoice ai
        WHERE ai.move_id = am.id
        """,
    )
</pre>